src/main/java/seedu/address/storage/CsvAppointmentArchive.java

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * A class to access archived appointment data stored as a csv file on the hard disk.
+ */
+public class CsvAppointmentArchive implements AppointmentArchive {
+
+    private static final Logger logger = LogsCenter.getLogger(CsvAppointmentArchive.class);
+
+    // Statistics
+    private int numberOfArchivedAppointments = 0;
+    private int numberOfExpiredAppointments = 0;
+
+    private final Path directoryPath;
+
+    public CsvAppointmentArchive(Path directoryPath) {
+        this.directoryPath = directoryPath;
+    }
+
+    public Path getArchiveDirectoryPath() {
+        return directoryPath;
+    }
+
+    @Override
+    public ReadOnlyAppointmentBook archivePastAppointments(ReadOnlyAppointmentBook appointmentBook) {
+        requireNonNull(appointmentBook);
+        List<Appointment> appointments = appointmentBook.getAppointmentList();
+
+        LocalDate date = null;
+        List<CsvAdaptedAppointment> pastAppointments = new ArrayList<>();
+        List<Appointment> upcomingAppointments = new ArrayList<>();
+
+        for (int i = 0; i < appointments.size(); i++) {
+            Appointment appointment = appointments.get(i);
+            LocalDate appointmentDate = appointment.getDate().getDate();
+
+            if (date == null || date != appointmentDate) {
+                if (DateTimeUtil.isExpiredByDay(appointmentDate)) {
+                    // change of day
+                    if (date != null && (date.getYear() != appointmentDate.getYear()
+                            || date.getMonth() != appointmentDate.getMonth())) {
+                        saveAppointments(pastAppointments, getFileName(date));
+                    }
+                    pastAppointments.clear();
+                    date = appointmentDate;
+                } else {
+                    // no more expired appointments
+                    while (i < appointments.size()) {
+                        upcomingAppointments.add(appointments.get(i));
+                        i++;
+                    }
+                    break;
+                }
+            }
+            pastAppointments.add(new CsvAdaptedAppointment(appointment));
+            numberOfArchivedAppointments++;
+            numberOfExpiredAppointments += appointment.getIsDoneStatus() ? 0 : 1;
+        }
+
+        if (pastAppointments.isEmpty()) {
+            return appointmentBook;
+        }
+
+        saveAppointments(pastAppointments, getFileName(date));
+
+        AppointmentBook book = new AppointmentBook();
+        book.setAppointments(upcomingAppointments);
+        return book;
+    }
+
+    @Override
+    public void saveAppointments(List<CsvAdaptedAppointment> appointments, String fileName) {
+        try {
+            saveAppointments(appointments, directoryPath.resolve(fileName));
+        } catch (IOException ioe) {
+            logger.warning("Failed to archive past appointment: " + ioe);
+        }
+    }
+
+    /**
+     * Same as {@link #saveAppointments(List, String)}.
```

Comment:
Should we paste the description here instead of give a reference?

Reply:
Agreed! 👍